### ISSUE TITLE
stv:player - links with all numbers in the metadata fail

### ISSUE THREAD

**Issue by james-wallace-ghub** [ytdl-org/youtube-dl#23083]
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2019.11.05**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-u', 'PRIVATE', '-p', 'PRIVATE', 'https://player.stv.tv/episode/3113/get-wet', '--verbose']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2019.11.05
[debug] Python version 3.4.4 (CPython) - Windows-8.1-6.3.9600
[debug] exe versions: none
[debug] Proxy map: {}
[stv:player] 3113: Downloading JSON metadata
ERROR: Unable to download JSON metadata: HTTP Error 404: Not Found (caused by HT
TPError()); please report this issue on https://yt-dl.org/bug . Make sure you ar
e using the latest version; type  youtube-dl -U  to update. Be sure to call yout
ube-dl with the --verbose flag and include its complete output.
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7hc5a0
9v\build\youtube_dl\extractor\common.py", line 627, in _request_webpage
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp7hc5a0
9v\build\youtube_dl\YoutubeDL.py", line 2237, in urlopen
  File "C:\Python\Python34\lib\urllib\request.py", line 470, in open
  File "C:\Python\Python34\lib\urllib\request.py", line 580, in http_response
  File "C:\Python\Python34\lib\urllib\request.py", line 508, in error
  File "C:\Python\Python34\lib\urllib\request.py", line 442, in _call_chain
  File "C:\Python\Python34\lib\urllib\request.py", line 588, in http_error_default
```

## Description

It seems that there is an issue with extracting from websites if all 4 characters in the URL are numbers. The example is one I've found that should be accessible to all with an account, that demonstrates the issue specifically.

To get an account from player.stv.tv, you will need proof of being in Scotland - a postcode from https://www.townscountiespostcodes.co.uk/postcodes-in-scotland/ should do it

**Comment by MatthewVernon** [comment]
I've also found this issue with youtube-dl and episodes with integers in.

**Comment by MatthewVernon** [comment]
@remitamine / @vangelis66 this bug doesn't seem to be fixed? Using the episode URL in the test for the new version:
```
[stv:player] 4125: Downloading JSON metadata
ERROR: Unable to download JSON metadata: HTTP Error 404: Not Found (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
matthew@aragorn:/tmp$ /tmp/youtube-dl --verbose -u 'bugmenot@mailinator.com' -p 'bugmenot' https://player.stv.tv/episode/4125/jennifer-saunders-memory-lane
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'--verbose', u'-u', u'PRIVATE', u'-p', u'PRIVATE', u'https://player.stv.tv/episode/4125/jennifer-saunders-memory-lane']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2021.01.03
[debug] Python version 2.7.13 (CPython) - Linux-4.9.0-12-amd64-x86_64-with-debian-9.13
[debug] exe versions: ffmpeg 3.2.15-0, ffprobe 3.2.15-0, rtmpdump 2.4
[debug] Proxy map: {}
[stv:player] 4125: Downloading JSON metadata
ERROR: Unable to download JSON metadata: HTTP Error 404: Not Found (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
  File "/tmp/youtube-dl/youtube_dl/extractor/common.py", line 632, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/tmp/youtube-dl/youtube_dl/YoutubeDL.py", line 2248, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python2.7/urllib2.py", line 435, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 548, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 473, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 556, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
```

This is using the latest version (I updated this morning), and I double-checked both that the credentials work (typed them into a web-browser) and that in general downloading works (I picked a random youtube video) and that the video is playable from the web.

**Comment by remitamine** [comment]
as been stated in the other issue, it will be included in the next version.

**Comment by MatthewVernon** [comment]
Oh, sorry, I got the github notification that the issue was fixed and saw that there was a new version out and assumed the fix was in 2021.01.03.
